Documentation/install/upgrade.rst

```diff
@@ -277,6 +277,52 @@ Changes that may require action
    reduced binary size. If you want to continue using the option, please use an
    older version of the cilium-bugtool binary.
 
+.. _1.4_new_options:
+
+New ConfigMap Options
+~~~~~~~~~~~~~~~~~~~~~
+
+  * ``enable-ipv4``: replaces ``disable-ipv4``. If enabled, all endpoints are
+    allocated an IPv4 address.
+
+  * ``enable-ipv6``: If enabled, all endpoints are allocated an IPv6 address.
+
+  * ``preallocate-bpf-maps``: If true, reduce per-packet latency at the expense
+    of up-front memory allocation for entries in BPF maps. If this value is
+    modified, then during the next Cilium startup the restore of existing
+    endpoints and tracking of ongoing connections may be disrupted.This may
+    lead to policy drops or a change in loadbalancing decisions for a
+    connection for some time. Endpoints may need to be recreated to restore
+    connectivity. If this option is set to "false" during an upgrade from 1.3
+    or earlier to 1.4 or later, then it may cause one-time disruptions during
+    the upgrade.
+
+  * ``flannel-master-device``: When running Cilium with policy enforcement
+    enabled on top of a CNI plugin, the BPF programs will be installed on the
+    network interface specified in this option and on each network interface
+    belonging to a pod.
+
+  * ``flannel-uninstall-on-exit``: If ``flannel-master-device`` is specified,
+    this determines whether Cilium should remove BPF programs from the master
+    device and interfaces belonging to pods when the Cilium `DaemonSet` is
+    deleted. If true, Cilium will remove programs from the pods.
+
+  * ``flannel-manage-existing-containers``: On startup, installs a BPF program
```

Comment:
Are all of these options intended to stay as `flannel-xxx`, given that there's already talk of integrating with other CNI? Should we rename these first?

Reply:
No, they are intended to be kept with that name.